src/store/hotspotDetails/hotspotDetailsSlice.ts

```diff
@@ -72,66 +33,135 @@ const fillMissingHourlyValues = (array: Sum[]) => {
+  )
+  const witnessAverage = totalAvg.avg / witnessSums.length
+  const witnessAveragePrev = prevAvg.avg / witnessSumsPrevious.length
+  return {
+    witnessSums,
+    witnessAverage,
+    witnessChange: calculatePercentChange(witnessAverage, witnessAveragePrev),
+  }
+}
+
+export const fetchHotspotChallengeSums = async (
+  params: FetchDetailsParams,
+  today: Date,
+  previousMaxDate: Date,
+  bucket: Bucket,
+) => {
+  const challengeSums = await getHotspotChallengeSums({
+    address: params.address,
+    minTime: `-${params.numDays} day`,
+    maxTime: today,
+    bucket,
+  })
+  const challengeSumsPrevious = await getHotspotChallengeSums({
+    address: params.address,
+    minTime: `-${params.numDays} day`,
+    maxTime: previousMaxDate,
+    bucket,
+  })
+  challengeSums.reverse()
+  const totalSum = challengeSums.reduce(
+    (a, b) => ({ sum: a.sum + b.sum } as any),
+  )
+  const prevSum = challengeSumsPrevious.reduce(
+    (a, b) => ({ sum: a.sum + b.sum } as any),
+  )
+  return {
+    challengeSums,
+    challengeSum: totalSum.sum,
+    challengeChange: calculatePercentChange(totalSum.sum, prevSum.sum),
+  }
+}
+
+export const fetchHotspotDetails = createAsyncThunk(
+  'hotspotDetails/fetchHotspotDetails',
+  async (params: FetchDetailsParams) => {
     const bucket = params.numDays === 1 ? 'hour' : 'day'
-    const today = new Date()
-    const previousMaxDate = new Date(today)
-    previousMaxDate.setDate(previousMaxDate.getDate() - params.numDays)
-    const witnessSums = await getHotspotWitnessSums({
-      address: params.address,
-      minTime: `-${params.numDays} day`,
-      maxTime: today,
-      bucket,
-    })
-    const witnessSumsPrevious = await getHotspotWitnessSums({
-      address: params.address,
-      minTime: `-${params.numDays} day`,
-      maxTime: previousMaxDate,
-      bucket,
-    })
-    if (bucket === 'hour') {
-      fillMissingHourlyValues(witnessSums)
-      fillMissingHourlyValues(witnessSumsPrevious)
-    }
-    witnessSums.reverse()
-    const totalAvg = witnessSums.reduce(
-      (a, b) => ({ avg: a.avg + b.avg } as any),
-    )
-    const prevAvg = witnessSumsPrevious.reduce(
-      (a, b) => ({ avg: a.avg + b.avg } as any),
-    )
-    const witnessAverage = totalAvg.avg / witnessSums.length
-    const witnessAveragePrev = prevAvg.avg / witnessSumsPrevious.length
+    const startDate = new Date()
+    const endDate = new Date(startDate)
+    endDate.setDate(endDate.getDate() - params.numDays)
+    const data = await Promise.all([
+      getHotspotDetails(params.address),
+      getHotspotRewardsSum(params.address, params.numDays),
+      getHotspotRewardsSum(params.address, params.numDays, endDate),
+      getHotspotRewards(params.address, params.numDays),
+      getHotspotWitnesses(params.address),
+      fetchHotspotWitnessSums(params, startDate, endDate, bucket),
+      fetchHotspotChallengeSums(params, startDate, endDate, bucket),
+    ])
+    // TODO: handle failures
```

Comment:
Don't know what we want to do about failures? Right now it will just show no data and if you change the timeline it would try to load again